ssm/lds.py

```diff
@@ -609,52 +659,129 @@ def hessian_neg_expected_log_joint(x, Ez, Ezzp1, scale=1):
 
             # Use Newton's method or LBFGS to find the argmax of the expected log joint
             scale = x0.size
-            obj = lambda x: neg_expected_log_joint(x, Ez, Ezzp1, scale=scale)
+
+            def obj(x):
+                return self._laplace_neg_expected_log_joint(data,
+                                                            input,
+                                                            mask,
+                                                            tag,
+                                                            x,
+                                                            Ez,
+                                                            Ezzp1,
+                                                            scale=scale)
             if optimizer == "newton":
                 # Run Newtons method
-                grad_func = lambda x: grad_neg_expected_log_joint(x, Ez, Ezzp1, scale=scale)
-                hess_func = lambda x: hessian_neg_expected_log_joint(x, Ez, Ezzp1, scale=scale)
+
+                def grad_func(x):
+                    return grad_neg_expected_log_joint(data,
```

Comment:
Lots of redundant lines of code here.  I'd propose something like:
```
kwargs = dict(data=data, input=input, mask=mask, tag=tag, Ez=Ez, Ezzp1=Ezzp1, scale=scale)
obj = lambda x: self._laplace_neg_expected_log_joint(x=x, **kwargs)
grad_obj = lambda x: grad(self._laplace_neg_expected_log_joint, argnum=4)(x=x, **kwargs)
hess_obj = lambda x: self. _laplace_hessian_neg_expected_log_joint(x=x, **kwargs)
```
and then below in the LBFGS block call 
```
lbfgs(lambda x, itr: obj(x), ...)
```

Reply:
I like using kwargs as suggested -- my pep8 checker complains about assigning the value of a `lambda` though (apparently it's not pep8 compliant to assign the output of a lambda expression) so I might replace those with a one-line `def`.

Reply:
Oops.  Didn't realize that wasn't allowed.  Good to know.